src/__tests__/test-utils-wrapper.test.tsx

```diff
@@ -0,0 +1,99 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import fs from 'fs';
+import path from 'path';
+import pluralize from 'pluralize';
+
+import * as components from '../../lib/components';
+import * as dom from '../../lib/components/test-utils/dom';
+import * as selectors from '../../lib/components/test-utils/selectors';
+
+jest.mock('@cloudscape-design/test-utils-core/dom');
+jest.mock('@cloudscape-design/test-utils-core/selectors');
+
+function getComponentsList() {
+  const exceptions: Record<string, string> = {
+    AnnotationContext: 'Annotation',
+  };
+
+  return Object.keys(components)
+    .map(component => exceptions[component] ?? component)
+    .filter(Boolean);
+}
+
+describe('Test utils element wrapper', () => {
+  describe('Snapshot tests', () => {
+    const importPaths = [
+      {
+        type: 'dom',
+        relativePath: '../../lib/components/test-utils/dom/index.js',
+      },
+      {
+        type: 'selectors',
+        relativePath: '../../lib/components/test-utils/selectors/index.js',
+      },
+    ] as const;
+
+    test.each(importPaths)('$type wrapper stays unchanged', ({ relativePath }) => {
+      const testUtilsPath = path.join(__dirname, relativePath);
+      const domWrapper = fs.readFileSync(testUtilsPath, 'utf8');
+      expect(domWrapper).toMatchSnapshot();
+    });
+  });
+
+  describe.each(getComponentsList())('<%s />', componentName => {
+    beforeEach(() => {
+      jest.clearAllMocks();
+    });
+
+    const testUtilTypes = [
+      {
+        type: 'dom',
+        testWrapper: new dom.ElementWrapper(document.createElement('div')),
+        testUtilImport: dom,
+      },
+      {
+        type: 'selector',
+        testWrapper: new selectors.ElementWrapper('div'),
+        testUtilImport: selectors,
+      },
+    ] as const;
+
+    describe.each(testUtilTypes)('$type wrapper', ({ testWrapper, testUtilImport }) => {
+      const componentNamePluralized = pluralize(componentName);
+
+      test(`find${componentName} calls findComponents with ${componentName}'s root selector`, () => {
+        const WrapperClass = testUtilImport[`${componentName}Wrapper` as 'AlertWrapper'];
```

Comment:
Why as `AlertWrapper`? There's nothing special about `AlertWrapper` just that the key is generated dynamically and to keep TS satisfied, we cast it to one of the components arbitrarily. For the purpose of this test file, all of the wrappers must have the exact same behavior.